### ISSUE TITLE
Allow using bun instead of node

### ISSUE THREAD

**Issue by gaborbernat** [pre-commit/pre-commit#3089]
### search you tried in the issue tracker

bun

### describe your actual problem

Documentation about bun: https://bun.sh.

The primary benefit would be much faster startup/run on larger projects.

### pre-commit --version

latest

**Comment by gaborbernat** [comment]
I'll try if I get to it, no promises 😆 would it be possible to convince https://github.com/pre-commit/mirrors-prettier#prettier-mirror to use bun without needing changes to that repo, though?

**Comment by asottile** [comment]
yeah in theory you'd set `language: bun` and it should just work assuming we make `language: bun` as compatible with `language: node` as possible

**Comment by gaborbernat** [comment]
So it would be:
```
-   repo: https://github.com/pre-commit/mirrors-prettier
    language: bun
    rev: ''  # Use the sha / tag you want to point at
    hooks:
    -   id: prettier
```
?

**Comment by matanshavit** [comment]
I gave it a shot [#3568](https://github.com/pre-commit/pre-commit/pull/3568)

**Comment by gaborbernat** [comment]
There does not seem to be enough interest in this, so I am closing it. Thanks for taking a look.
